/**
 * Copyright 2022 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package mindspore.distributed.cluster.topology;

message RegistrationMessage {
  string node_id = 1;
  string host_name = 2;
  string role = 3;
  string host_ip = 4;
}

message RegistrationRespMessage {
  bool success = 1;
  uint32 rank_id = 2;
  uint32 node_num = 3;
  string error_reason = 4;
}

message UnregistrationMessage {
  string node_id = 1;
}

message HeartbeatMessage {
  string node_id = 1;
}

message HeartbeatRespMessage {
  bool success = 1;
  uint32 topo_state = 2;
  uint32 nodes_num = 3;
  uint32 abnormal_nodes_num = 4;
}

message MetadataMessage {
  string name = 1;
  bytes value = 2;
}

message ActorAddress {
  string actor_id = 1;
  string ip = 2;
  uint32 port = 3;
  uint32 func_id = 4;
}

message PortRange {
  uint32 min_port = 1;
  uint32 max_port = 2;
}

message NodePortRanges {
  map<string, PortRange> data = 1;
}
